online_meetings/canvas_zoom_meetings.py

```diff
@@ -67,42 +72,61 @@ def get_zoom_json(self, page_num: int = 1) -> Dict:
             return zoom_json["result"]
         return None
 
+    def extract_from_js(self, key: str, text: str) -> Optional[str]:
+        """Takes a javascript text and attempts to extract a key/value (May not work with everything)
+        
+        :param key: key to extract
+        :type key: str
+        :param text: text to search in
+        :type text: str
+        :return: The matching string or None if not found
+        :rtype: Optional[str]
+        """
+        pattern = re.search(f'{key}.*"(.*)"', text)
```

Comment:
This seems to be doing what you want it to. I wish there was a way we could just parse the string as JSON, as that seems less error-prone than a somewhat-hard-to-read regular expression. However, looking at one example, it looks like there are some single quotes in the `<script>` block, so you might have to do some string cleaning before you could successfully parse it.

Reply:
Yeah, I tried to research this a bit but the problem is that this isn't JSON this is Javascript coming back so it's not as parsable a language. I didn't find any great updated package that could parse javascript in Python and most SO suggestions just recommended using Regex. 

There's stuff out there like https://github.com/rspivak/slimit and https://github.com/PiotrDabkowski/pyjsparser to lex Javascript but they're out of date and probably too much. I would like to make this better but not spend too much time on it since I currently only need two specific keys. Maybe something to keep in mind.